lib/manageiq/appliance_console/external_auth_options.rb

```diff
@@ -79,11 +81,44 @@ def update_configuration(update_hash = nil)
       if update_hash.present?
         say("\nUpdating external authentication options on appliance ...")
         params = update_hash.collect { |key, value| "#{key}=#{value}" }
+        params = enable_provider_type(params)
         result = ManageIQ::ApplianceConsole::Utilities.rake_run("evm:settings:set", params)
         raise parse_errors(result).join(', ') if result.failure?
       end
     end
 
+    def validate_provider_type
+      return true unless @updates["/authentication/oidc_enabled"] == true && @updates["/authentication/saml_enabled"] == true
+      say("\Error: Both SAML and OIDC can not be enabled ...")
+      false
+    end
+
+    def enable_provider_type(params)
+      if params.include?("/authentication/saml_enabled=true")
+        provider_type_saml(params)
+      elsif params.include?("/authentication/oidc_enabled=true")
+        provider_type_oidc(params)
+      elsif params.include?("/authentication/oidc_enabled=false") || params.include?("/authentication/saml_enabled=false")
+        provider_type_none(params)
+      end
+    end
+
+    def provider_type_saml(params)
```

Comment:
I think I would prefer these have a method name that uses a verb as it's altering the hash. A noun method feels more like it's returning a value retrieved from somewhere else.

Also all of these should be named with a `!` for the same reason as above.